lib/xandra/connection.ex

```diff
@@ -32,17 +38,19 @@ defmodule Xandra.Connection do
 
   def handle_prepare(%Prepared{} = prepared, options, %__MODULE__{socket: socket} = state) do
     force? = Keyword.get(options, :force, false)
+    compressor = assert_valid_compressor(state.compressor, options[:compressor])
```

Comment:
Then we introduce the nastyness of having to find a good variable name for `compressor` because we have to differentiate this `compressor` and `state.compressor`. Suggestions welcome.